fakebook/src/app/component/post-view/post-view.component.spec.ts

```diff
@@ -1,19 +1,54 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ComponentFixture, TestBed, fakeAsync, waitForAsync, inject } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { NavigationBehaviorOptions , Router } from '@angular/router';
+import { NEVER } from 'rxjs';
 
+import { User } from '../../model/user';
+import { Comment } from '../../model/comment';
+import { Post } from '../../model/post';
 import { PostViewComponent } from './post-view.component';
 
 describe('PostViewComponent', () => {
   let component: PostViewComponent;
   let fixture: ComponentFixture<PostViewComponent>;
 
+  const testUser: User = {
+    id: 1,
+    firstName: 'first',
+    lastName: 'last',
+    email: 'e@mail',
+    phoneNumber: undefined,
+    profilePictureUrl: null,
+    status: undefined,
+    birthDate: new Date(2010, 12),
+  };
+
+  const testComment: Comment = {
+    id: 1,
+    userId: 1,
+    content: 'comment content',
+    postId: 1,
+    createdAt: undefined,
+    user: undefined
+  };
+
+  const testPost: Post = {
+    id: 1,
+    content: 'string',
+    user: testUser,
+    pictureUrl: undefined,
+    createdAt: new Date(2020, 12),
+    likedByUserIds: [],
+    commentIds: [1],
+    comments: [testComment],
+    liked: false
+  };
+
```

Comment:
This block is what causing the sonar gate to fail. but It shouldn't be an issue here for now. gonna approve so we can work on implementation